### PR TITLE
Scaffold out KubernetesCluster (+ some initial implementation)

### PR DESCRIPTION
**Pull request by nemacysts**
This sets up the skeleton for KubernetesCluster (a k8s version of
MesosCluster) and does some of the basic implementation for several
methods when possible/easy

### CODE REVIEW

File: tron/kubernetes.py [hunk 1]
```diff
@@ -91,3 +104,206 @@ def handle_event(self, event: Event) -> None:
...
+
+    def handle_next_event(self, _=None) -> None:
+        """
+        Pops events off of the shared tron<->task_proc queue and processes them.
+
+        We only expect two types of events:
+        * control: events regarding how the task_proc plugin is running - handled directly
+        * task: events regarding how the actual tasks/Pods we're running our doing - forwarded to KubernetesTask
+        """
+        if self.deferred and not self.deferred.called:
```
Comment (lanceleih) [inline tron/kubernetes.py:183]: Are we checking if self.deferred has been called? I'd assume it to be true if we already have handlers waiting. When does self.deferred.called become true?

Comment (nemacysts) [reply tron/kubernetes.py:183]: self.deferred will either be `None` or a `PyDeferred` object (thus the `if self.deferred`), and until one of the {Call,Err}backs has been called, `self.deferred.call` will be `False` - in this case, `called` will be `False` until `process_event()` or `logError` have been called

File: tests/kubernetes_test.py [hunk 2]
```diff
@@ -73,3 +79,74 @@ def test_handle_event_exit_early_on_misrouted_event(mock_kubernetes_task):
...
+
+def test_stop_default(mock_kubernetes_cluster):
+    # When stopping, tasks should not exit. They will be recovered
+    mock_task = mock.MagicMock()
+    mock_kubernetes_cluster.tasks = {"task_id": mock_task}
+    mock_kubernetes_cluster.stop()
+    assert mock_kubernetes_cluster.deferred is None
+    assert mock_task.exited.call_count == 0
+    assert len(mock_kubernetes_cluster.tasks) == 1
+
```
Comment (lanceleih) [inline tests/kubernetes_test.py:147]: Can we also test set_enabled? Then we can check if connect() and stop() are being called properly.

Comment (nemacysts) [reply tests/kubernetes_test.py:161]: done!

**Review by lanceleih — APPROVED** [review]
LGTM!

**Pull request merged**